string token_address
string did
string pool_address